pkg/operator/operator.go

```diff
@@ -126,6 +129,15 @@ func New(config operatorconfig.Config, kubeConfig *rest.Config) (*Operator, erro
 		}
 	}
 
+	// verify that all idled services have the correct idle annotations
+	// mirrored over from the corresponding endpoints resources.
+	// This is to ensure that applications idled with an older version of oc
+	// (and thus do not have the idle annotations on the service) are still
+	// safely un-idleable afer an upgrade that affects `oc idle` functionality.
+	if err := ensureServicesHaveIdleAnnotation(mgr.GetClient()); err != nil {
```

Comment:
Using `mgr.GetClient()` before the manager `Start` has been called (and thus caches synched) makes me a little nervous...  But I think this is all right because we explicitly specified `NewClient` not to use a caching client in the manager options.

Reply:
> But I think this is all right because we explicitly specified `NewClient` not to use a caching client in the manager options.

Is there an alternative code path you'd like to see instead? I agree that since we are using a cacheless client anyways this is probably fine as is.

Reply:
Use a separate client (you can use either the controller-runtime or client-go package to get the client, but it should be a separately instantiated client).  

I'd go so far as to suggesting divorcing this migration logic from the controller-runtime logic.  Put it in `start.go`, or even put it in a separate subcommand, invoked using an init container.

Reply:
Alright, Ill move this logic to start.go and create a separate client there.